### ISSUE TITLE
NaN value happens in the middle of training

### ISSUE THREAD

**Issue by Beniko95J** [yenchenlin/nerf-pytorch#82]
Hi, thanks for the re-implementation of NeRF in pytorch. I am trying to run this repo on my own dataset, but NaN value happens in the middle of training. This confuses me since the training looks good in the start. Has anyone met this issue before?

![image](https://user-images.githubusercontent.com/30772597/185024959-77bac05a-71ec-4955-a2df-45fa8b3d2051.png)

**Comment by oakshy** [comment]
I also met the same problem. Did you solve it now?

**Comment by yuanyue** [comment]
Same question here.

**Comment by YuvalMandel** [comment]
I have the same problem, Did anyone manage to solve it?

**Comment by yuanyue** [comment]
> I have the same problem, Did anyone manage to solve it?

I reduce the learning rate and it works.  You can try it.

**Comment by YuvalMandel** [comment]
Thanks! I tried to reduce from 1e-6 to 1e-9 and also 1e-12, but it doesn't see, to be doing much of a difference. I will also add that the NaN's don't always appear at the start, and usually appear in later iterations.

**Comment by chinnarouge** [comment]
> > I have the same problem, Did anyone manage to solve it?
> 
> I reduce the learning rate and it works. You can try it.

could you please tell how you reduced the learning rate  and how to reduce the number of iterations

**Comment by silver-obelisk** [comment]
I increased factor,and the problem was sloved.But it is not a good way to increase factor,that means low resolution

**Comment by silver-obelisk** [comment]
> > I have the same problem, Did anyone manage to solve it?
> 
> I reduce the learning rate and it works. You can try it.

I wonder what lr you set?  I change 5e-4 to 5e-5 but unuseful.
